scripts/misc/update_match_seq_region_ids.pl

```diff
@@ -0,0 +1,181 @@
+# limitations under the License.
+
+use strict;
+use warnings;
+
+use Bio::EnsEMBL::Registry;
+
+use DBI qw(:sql_types);
+use Getopt::Long;
+
+usage() if (!scalar(@ARGV));
+
+my $config = {};
+
+GetOptions(
+  $config,
+  'registry=s',
+  'dry_run',
+  'help!',
+) or die "Error: Failed to parse command line arguments\n";
+
+usage() if ($config->{help});
+
+die ('A registry file is required (--registry)') unless (defined($config->{registry}));
+
+my $registry = 'Bio::EnsEMBL::Registry';   
+$registry->load_all($config->{registry});
+
+my $cdbas = $registry->get_all_DBAdaptors(-group => 'core');
+my $vdbas_tmp = $registry->get_all_DBAdaptors(-group => 'variation');
+my $vdbas = {};
+foreach my $vdba (@$vdbas_tmp) {
+  my $species = $vdba->species;
+  $vdbas->{$species} = $vdba;
+}
+
+my @tables = (
+  "compressed_genotype_region",
+  "phenotype_feature",
+  "read_coverage",
+  "variation_feature",
+  "structural_variation_feature"
+  );
+
+foreach my $cdba (@$cdbas) {
+  my $species = $cdba->species;
+  die ("Species not found") if (!$vdbas->{$species});
+  my $dbh = $cdba->dbc->db_handle;
+  my $cd_dbname = $cdba->dbc->dbname;
+  my $id_mapping = {};
+  my $sth = $dbh->prepare("SELECT seq_region_id, name FROM seq_region;");
+  $sth->execute();
+  while (my @row = $sth->fetchrow_array) {
+    my $external_seq_region_id = $row[0];
+    my $external_seq_region_name = $row[1];
+    $id_mapping->{$external_seq_region_name} = $external_seq_region_id;     
+  }
+  $sth->finish();
+  
+  my $vdba = $vdbas->{$species};
+  my $vdbh = $vdba->dbc->db_handle;
+
+  my $vd_dbname = $vdba->dbc->dbname;
+  my $vd_mapping = {};
+  $sth = $vdbh->prepare("SELECT seq_region_id, name FROM seq_region;");
+  $sth->execute();
+  while (my @row = $sth->fetchrow_array) {
+    my $internal_seq_region_id = $row[0];
+    my $internal_seq_region_name = $row[1];
+
+    # Check if Variation and Core are equal in a name
+    die "ERROR: '$internal_seq_region_name' is not listed in $cd_dbname.seq_region name column\n." unless defined($id_mapping->{$internal_seq_region_name});
+    next if (defined($id_mapping->{$internal_seq_region_name}) && $id_mapping->{$internal_seq_region_name} eq $internal_seq_region_id);
+
+    $vd_mapping->{$internal_seq_region_name} = $internal_seq_region_id;
+  }
+  $sth->finish();
+
+  # Core vs. Variation seq_region_id's
+  die ("All seq_region_id's has a correct match - Variation vs. Core\n") if (defined($vd_mapping));
+
+  # Remove old seq_region_id from vdb and create a new one based on core
+  unless (defined($config->{dry_run})) {
+    $vdbh->do("ALTER TABLE seq_region drop seq_region_id") or die $dbh->errstr;
+    $vdbh->do("ALTER TABLE seq_region ADD seq_region_id INT") or die $dbh->errstr;
+    $vdbh->do("ALTER TABLE seq_region MODIFY COLUMN seq_region_id INT UNSIGNED NOT NULL FIRST") or die $dbh->errstr;
+    $vdbh->do("ALTER TABLE seq_region ADD PRIMARY KEY (seq_region_id)") or die $dbh->errstr;
```

Comment:
We don't need to drop and add the seq_region_id here. 
The seq_region table is being updated in line 135 - the old seq_region_ids are being replaced with new ones.

Reply:
Moved or removed some of those alter tables. It was not working properly. Now it will make seq_region_id, a primary key just after running whole update.